test/replication-luatest/gh_9918_synchro_queue_additional_info_test.lua

```diff
@@ -26,6 +26,7 @@ g.before_each(function(cg)
     cg.replica_set:start()
     cg.leader:exec(function()
         box.ctl.promote()
+        box.ctl.wait_rw()
```

Comment:
> The test expects `box.ctl.promote()` to be synchronous, i.e. that the
caller will wait until his node actually becomes the leader. But this is
not the case, the user must manually call `box.ctl.wait_rw()` to ensure
that he can actually write.

This test does not use elections. The instances are read-write by default, and AFAIC, the instance owns the limbo after `box.ctl.promote` returns:
https://github.com/tarantool/tarantool/blob/850673db5a69df2c7250d174ab15305624b2634a/src/box/box.cc#L3086

So I don't see any reason why the `leader` node should not be writable after `box.ctl.promote`.

It is not clear why the `leader` instance is orphan from the logs you presented in the issue. My guess is that it is because the replica set has not synced. If this is the case, then AFAIC, the right fix would be to move the `cg.replica_set:wait_for_fullmesh()` right after `cg.replica_set:start()`.

Reply:
Yes, you are right. Fixed.